packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -0,0 +1,128 @@
+      this.getOrCreateResource(logicalId);
+    }
+  }
+
+  /**
+   * Returns the low-level CfnResource from the template with the given logical ID.
+   * Any modifications performed on that resource will be reflected in the resulting CDK template.
+   *
+   * The returned object will be of the proper underlying class;
+   * you can always cast it to the correct type in your code:
+   *
+   *     // assume the template contains an AWS::S3::Bucket with logical ID 'Bucket'
+   *     const cfnBucket = cfnTemplate.getResource('Bucket') as s3.CfnBucket;
+   *     // cfnBucket is of type s3.CfnBucket
+   *
+   * If the template does not contain a resource with the given logical ID,
+   * an exception will be thrown.
+   *
+   * @param logicalId the logical ID of the resource in the CloudFormation template file
+   */
+  public getResource(logicalId: string): core.CfnResource {
+    const ret = this.resources[logicalId];
+    if (!ret) {
+      throw new Error(`Resource with logical ID '${logicalId}' was not found in the template`);
+    }
+    return ret;
+  }
+
+  /** @internal */
+  public _toCloudFormation(): object {
+    const ret: { [section: string]: any } = {};
+
+    for (const section of Object.keys(this.template)) {
+      // render all sections of the template unchanged,
+      // except Resources, which will be taken care of by the created L1s
+      if (section !== 'Resources') {
+        ret[section] = this.template[section];
+      }
+    }
+
+    return ret;
+  }
+
+  private getOrCreateResource(logicalId: string): core.CfnResource {
+    const ret = this.resources[logicalId];
+    if (ret) {
+      return ret;
+    }
+
+    const resourceAttributes: any = this.template.Resources[logicalId];
+    const l1ClassFqn = cfn_type_to_l1_mapping.lookup(resourceAttributes.Type);
+    if (l1ClassFqn) {
+      // fail early for resource attributes we don't support yet
+      const unsupportedAttributes = ['Condition', 'DependsOn', 'CreationPolicy', 'UpdatePolicy'];
+      for (const unsupportedAttribute of unsupportedAttributes) {
+        if (resourceAttributes[unsupportedAttribute]) {
+          throw new Error(`The ${unsupportedAttribute} resource attribute is not supported by cloudformation-include yet`);
+        }
+      }
+
+      const [moduleName, ...className] = l1ClassFqn.split('.');
+      const module = require(moduleName); // eslint-disable-line @typescript-eslint/no-require-imports
+      const jsClassFromModule = module[className.join('.')];
+      const l1Instance = jsClassFromModule.fromCloudFormation(this, logicalId, resourceAttributes);
+
+      if (this.preserveLogicalIds) {
+        // override the logical ID to match the original template
+        l1Instance.overrideLogicalId(logicalId);
+      }
+
+      this.resources[logicalId] = l1Instance;
+      return l1Instance;
+    } else {
+      throw new Error(`Unrecognized CloudFormation resource type: '${resourceAttributes.Type}'`);
+    }
```

Comment:
Why not just ignore these, and keep them in the output template?

Is this because we don't yet support unrecognized resource preservation? If so, can you add a comment.

Reply:
I think it would be incorrect to ignore these, and preserve them in the output (for example, `getResource()` on them would not work).

Added comment.